praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/service/PraktikumsstellenService.java

```diff
@@ -144,6 +123,35 @@ public List<StudiumsPraktikumsstelleDto> getAllAssignedStudiumspraktikumsstellen
                 .findAllByMeldezeitraumIDAndAssignedNwkIsNotNull(lastMeldezeitraumID).stream().map(praktikumsstellenMapper::toDto).toList();
     }
 
+    public TreeMap<String, List<PraktikumsstelleDto>> getAllInCurrentMeldezeitraumGroupedByDienststelle() {
+        final UUID currentMeldezeitraumID = meldezeitraumService.getCurrentMeldezeitraum().id();
+        return getPraktikumsstellenGroupedByDienststelle(currentMeldezeitraumID);
+    }
+
+    public TreeMap<String, List<PraktikumsstelleDto>> getRecentPraktikumsstellenGroupedByDienststelle() {
+        UUID lastMeldezeitraumID = meldezeitraumService.getMostRecentPassedMeldezeitraum().id();
+        return getPraktikumsstellenGroupedByDienststelle(lastMeldezeitraumID);
+    }
+
+    private TreeMap<String, List<PraktikumsstelleDto>> getPraktikumsstellenGroupedByDienststelle(UUID meldezeitraumID) {
```

Comment:
I think this could be public as it would be a valid use case to get all Praktikumsstellen for a specific meldezeitraum, but we could also change this the moment we need it.